docs/guides/packer-channel-management.md

```diff
@@ -49,3 +49,71 @@ resource "hcp_packer_channel_assignment" "alpine_advanced" {
 }
 ```
 
+## Automatically creating channels for new and existing buckets
+
+```terraform
+data "hcp_packer_bucket_names" "all" {}
+
+resource "hcp_packer_channel" "release" {
+  for_each = data.hcp_packer_bucket_names.all.names
+
+  name        = "release"
+  bucket_name = each.key
+}
+```
+
+Optionally, you can use Terraform functions/expressions to filter the list of bucket names before providing it to `for_each` if channels should only be created for a subset of buckets.
+
+### Managing channel assignment for automatically created channels
+
+Channel assignment for automatically created channels can be managed in many ways. A nonexhaustive example configuration is provided below. 
+
+The `iteration_id` attribute is used for example purposes, but any of the three iteration identifier attributes can be used.
+
+```terraform
+data "hcp_packer_iteration" "all_latest" {
+  for_each = data.hcp_packer_bucket_names.all.names
+
+  bucket_name = each.key
+  channel     = "latest"
+}
+
+data "hcp_packer_iteration" "bucket3_staging" {
+  bucket_name = "bucket3"
+  channel     = "staging"
+}
+
+resource "hcp_packer_channel_assignment" "release" {
+  for_each = merge(
+    { # Defaulting all channels to be unassigned and Terraform-managed
+      for bucketName in keys(hcp_packer_channel.release) :
+      bucketName => "none"
+    },
+    { # Assigning channels that match a filter to an iteration fetched from another channel
+      for bucketName in keys(hcp_packer_channel.release) :
+      bucketName => data.hcp_packer_iteration.all_latest[bucketName].id
+      if endswith(bucketName, "-dev")
+    },
+    { # Individual channel assignments
+      "bucket1" : "01H1SF9NWAK8AP25PAWDBGZ1YD"
+      "bucket2" : "01H28NJ7WPCZA0FZZ8G3FGGTAF"
+      "bucket3" : data.hcp_packer_iteration.bucket3_staging.id
+    }
+  )
+
+  bucket_name = each.key
+  # Using a reference for `channel_name` allows it to be generated dynamically
+  # in the `hcp_packer_channel.release` resource if desired.
+  channel_name = hcp_packer_channel.release[each.key].name
+
+  iteration_id = each.value
+}
+```
+
+You are not required to configure an assignment for every channel at the same time as automatic channel creation. Using Terraform builtin functions/expressions, it is possible to limit which channels should have assignments configured by default.
+
+The default value used in the example is `"none"`, which causes Terraform to set the channel to have no assigned iteration and continue to ensure that an iteration is not assigned elsewhere. Each of the three iteration identifier attributes has a "zero value" for this purpose: `iteration_id` and `iteration_fingerprint` both use `"none"`, and `iteration_version` uses `0`. The default value can also be set to `null` if you want to ensure that every channel has an assignment defined explicitly rather than using a default.
```

Comment:
Using a default value of `"none"`, all channel assignments are automatically adopted into Terraform management, but channels without an explicitly defined assignment will use that default value (`"none"`. 

Using a default value of `null`, channel assignments will not be automatically adopted into Terraform management, but Terraform will force the practitioner to explicitly set an assignment for each channel in the `for_each` map. This handles the case where both of these are true:
1. the user wants to mandate that assignments for all automatically created channels must be managed by Terraform
2. it is undesirable for channels to be given a default value

An example of where item 2 could apply: Assignments are being handled by a list of filters, and a bucket appears that does not match any of those filters. The user may not want the assignment to be defaulted to `"none"`, and would instead prefer to define a matching filter (or otherwise resolve the discrepancy) before the apply can succeed. In this case, removing the default entirely would solve the issue of not wanting to apply a default value, but would not allow the user to enforce the mandate in item 1. 

The TL;DR explanation is:
- No default
  - Equivalent to explicit assignment being "optional" (
   - If the user doesn't define an assignment for a bucket, assignment will not be managed by Terraform
- Default of `"none"`
  - Equivalent to explicit assignment being "optional" 
    - If the user doesn't define an assignment for a bucket, assignment is enforced as "unassigned" by Terraform 
- Default of `null`
  - Equivalent to explicit assignment being "required" 
    - If the user doesn't define an assignment for a bucket, throws an error

Reply:
Ohh got it! I was missing the adoption part when interpreting this one. Thanks for explaining!